apps/arweave/c_src/ar_mine_randomx.c

```diff
@@ -910,7 +1026,463 @@ static ERL_NIF_TERM randomx_reencrypt_chunk_nif(
 	return ok_tuple2(envPtr, reencryptedChunkTerm, decryptedChunkTerm);
 }
 
+static ERL_NIF_TERM randomx_encrypt_composite_chunk_nif(
+	ErlNifEnv* envPtr,
+	int argc,
+	const ERL_NIF_TERM argv[]
+) {
+	// RandomX rounds per sub-chunk.
+	int randomxRoundCount;
+	// RandomX iterations (randomxRoundCount each) per sub-chunk.
+	int iterations;
+	// The number of sub-chunks in the chunk.
+	int subChunkCount;
+	int jitEnabled, largePagesEnabled, hardwareAESEnabled;
+	struct state* statePtr;
+	ErlNifBinary inputData;
+	ErlNifBinary inputChunk;
+
+	if (argc != 9) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_get_resource(envPtr, argv[0], stateType, (void**) &statePtr)) {
+		return error(envPtr, "failed to read state");
+	}
+	if (!enif_inspect_binary(envPtr, argv[1], &inputData)) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_inspect_binary(envPtr, argv[2], &inputChunk) ||
+		inputChunk.size == 0 ||
+		inputChunk.size > MAX_CHUNK_SIZE) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_get_int(envPtr, argv[3], &jitEnabled)) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_get_int(envPtr, argv[4], &largePagesEnabled)) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_get_int(envPtr, argv[5], &hardwareAESEnabled)) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_get_int(envPtr, argv[6], &randomxRoundCount)) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_get_int(envPtr, argv[7], &iterations) ||
+		iterations < 1) {
+		return enif_make_badarg(envPtr);
+	}
+	if (!enif_get_int(envPtr, argv[8], &subChunkCount) ||
```

Comment:
Do we also need to test that `subChunkCount` is a multiple of 64? e.g. I believe a subChunkCount of 2 would pass these checks but would then run afoul of this comment `// Both MAX_CHUNK_SIZE and subChunkCount are multiples of 64`

Reply:
Or, maybe that comment just needs to be updated - since we don't care if subChunkCount is a multiple of 64 so long as all sub chunks are the same size and the sub-chunks are multiple of 64